src/Medidata.ZipkinTracer.HttpModule/ZipkinRequestContextModule.cs

```diff
@@ -13,37 +12,39 @@ public class ZipkinRequestContextModule : IHttpModule
         public void Init(HttpApplication context)
         {
             context.BeginRequest += (sender, args) =>
-                {
-                    string url = HttpContext.Current.Request.Path;
+            {
+                var zipkinConfig = new ZipkinConfig();
 
-                    var zipkinConfig = new ZipkinConfig();
+                var traceProvider = new TraceProvider(new HttpContextWrapper(HttpContext.Current), zipkinConfig.DontSampleListCsv, zipkinConfig.ZipkinSampleRate);
+                var logger = LogManager.GetLogger(GetType());
 
-                    var traceProvider = new TraceProvider(new System.Web.HttpContextWrapper(HttpContext.Current), zipkinConfig.DontSampleListCsv, zipkinConfig.ZipkinSampleRate);
-                    var logger = LogManager.GetLogger(this.GetType());
+                ITracerClient zipkinClient = new ZipkinClient(traceProvider, logger);
+                var span = zipkinClient.StartServerTrace(HttpContext.Current.Request.Url, HttpContext.Current.Request.HttpMethod);
 
-                    ITracerClient zipkinClient = new ZipkinClient(traceProvider, url, logger);
-                    zipkinClient.StartServerTrace();
-
-                    HttpContext.Current.Items["zipkinClient"] = zipkinClient;
-                };
+                HttpContext.Current.Items["zipkinClient"] = zipkinClient;
+                HttpContext.Current.Items["zipkinSpan"] = span;
+            };
 
             context.EndRequest += (sender, args) =>
-                {
-                    var zipkinClient = (ITracerClient)HttpContext.Current.Items["zipkinClient"];
-                    zipkinClient.EndServerTrace();
-                };
+            {
+                var zipkinClient = (ITracerClient)HttpContext.Current.Items["zipkinClient"];
+                var span = (Span)HttpContext.Current.Items["zipkinSpan"];
+                zipkinClient.EndServerTrace(span);
+            };
 
             context.Error += (sender, args) =>
-                {
-                    var zipkinClient = (ITracerClient)HttpContext.Current.Items["zipkinClient"];
-                    zipkinClient.EndServerTrace();
-                };
+            {
```

Comment:
I think we had better delete the method as a zipkin trace is useless on error handler and we shouldn't implement heavy process on error handler.